### PR TITLE
use go modules

### PR DESCRIPTION
**Pull request by greut**
Jumping over to go modules, let's depreciate the _gopkg.in_ way of fetching dependencies.

- [x] https://github.com/stretchr/testify/issues/745
- [x] adapt the readme

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module github.com/editorconfig/editorconfig-core-go
```
Comment (andreynering) [inline go.mod]: This patch will break stuff for current users unless we bump the version to v2 and tag this as v2.0.0.

```suggestion
module github.com/editorconfig/editorconfig-core-go/v2
```

Comment (greut) [reply go.mod]: Correct!

Comment (mstruebing) [reply go.mod]: I think we should do that.
There is nothing better to come than go modules in the near future.

**Comment by greut** [comment]
@andreynering any chances this gets moved forward?

**Comment by andreynering** [comment]
@greut Yeah, sorry. I have forgot about this.

Thanks for reminding me.

**Comment by andreynering** [comment]
Just published this as v2.0.0

**Pull request merged**